src/region.js

```diff
@@ -288,16 +240,22 @@ var Region = MarionetteObject.extend({
       this._restoreEl();
     }
 
-    if (!preventDestroy) {
-      this._destroyView();
+    if (!shouldPreventDestroy) {
+      this._destroyView(options);
     }
     this.triggerMethod('empty', view);
 
     // Remove region pointer to the currentView
     delete this.currentView;
 
-    if (preventDestroy) {
+    if (shouldPreventDestroy) {
```

Comment:
I think this, and maybe even the delete above this should go before the `triggerMethod('empty')`
FWIW, this could match show more and look like:
```
if(shouldPreventDestroy){
  this._detachView(view, options);
} else {
  this._destroyView(view, options);
}
```